utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -515,6 +574,23 @@ mod test {
         new_year_on_or_before_fixed_date::<Dangi>(fixed, prev_solstice).0
     }
 
+    #[test]
+    fn test_month_structure() {
+        // Mostly just tests that the assertions aren't hit
+        for year in 1900..2050 {
+            let fixed = crate::iso::fixed_from_iso(year, 1, 1);
+            let chinese_year = chinese_based_date_from_fixed::<Chinese>(fixed);
+            let (month_ends, leap) = month_structure_for_year::<Chinese>(
+                chinese_year.year_bounds.new_year,
+                chinese_year.year_bounds.next_new_year,
+            );
+            println!(
+                "{year} (chinese {}): {month_ends:?} {leap:?}",
+                chinese_year.year
+            );
```

Comment:
Suggestion: also consider asserting the output here versus

1. `pub fn month_days<C: ChineseBased>(year: i32, month: u8) -> u8`
2. `pub fn days_in_provided_year<C: ChineseBased>(year: i32) -> u16` (the sum of the month structure should equal the number of days in the year)